### PR TITLE
Fixed phpdoc types - 

### PR DESCRIPTION
**Pull request by fashxp**
in preparation for #13869

### CODE REVIEW

File: bundles/EcommerceFrameworkBundle/CartManager/CartPriceCalculatorInterface.php [hunk 1]
```diff
@@ -41,7 +41,8 @@ public function initModificators();
 
     /**
      * Calculates cart sums and saves results
-     *
+
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/CartManager/CartPriceCalculatorInterface.php]:
```suggestion
     *
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/AssetId.php [hunk 2]
```diff
@@ -19,6 +19,11 @@
 
 class AssetId implements InterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
+     * @return ?int
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/AssetId.php]:
```suggestion
     * @param array|null $config
     * @return int|null
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/DefaultObjects.php [hunk 3]
```diff
@@ -19,6 +19,11 @@
 
 class DefaultObjects implements RelationInterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/DefaultObjects.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/DefaultRelations.php [hunk 4]
```diff
@@ -21,6 +21,11 @@
 
 class DefaultRelations implements RelationInterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/DefaultRelations.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/DefaultStructuredTable.php [hunk 5]
```diff
@@ -23,6 +23,11 @@ class DefaultStructuredTable implements InterpreterInterface
 {
     use OptionsResolverTrait;
 
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/DefaultStructuredTable.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/IdList.php [hunk 6]
```diff
@@ -23,6 +23,11 @@ class IdList implements InterpreterInterface
 {
     use OptionsResolverTrait;
 
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/IdList.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/InterpreterInterface.php [hunk 7]
```diff
@@ -17,5 +17,10 @@
 
 interface InterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/InterpreterInterface.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/Numeric.php [hunk 8]
```diff
@@ -17,6 +17,11 @@
 
 class Numeric implements InterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/Numeric.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/ObjectId.php [hunk 9]
```diff
@@ -19,6 +19,11 @@
 
 class ObjectId implements InterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/ObjectId.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/ObjectIdSum.php [hunk 10]
```diff
@@ -19,6 +19,11 @@
 
 class ObjectIdSum implements InterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/ObjectIdSum.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/Soundex.php [hunk 11]
```diff
@@ -17,6 +17,11 @@
 
 class Soundex implements InterpreterInterface
 {
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/Soundex.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/Interpreter/StructuredTable.php [hunk 12]
```diff
@@ -22,6 +22,11 @@ class StructuredTable implements InterpreterInterface
 {
     use OptionsResolverTrait;
 
+    /**
+     * @param mixed $value
+     * @param ?array $config
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Interpreter/StructuredTable.php]:
```suggestion
     * @param array|null $config
```

File: bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql.php [hunk 13]
```diff
@@ -553,6 +553,12 @@ public function getGroupByRelationValues($fieldname, $countValues = false, $fiel
         }
     }
 
+    /**
+     * @param bool $excludeConditions
+     * @param ?string $excludedFieldname
+     * @param ?string $variantMode
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql.php]:
```suggestion
     * @param string|null $excludedFieldname
     * @param string|null $variantMode
```

File: bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql.php [hunk 14]
```diff
@@ -626,6 +632,10 @@ protected function buildQueryFromConditions($excludeConditions = false, $exclude
         return $condition;
     }
 
+    /**
+     * @param ?string $excludedFieldname
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql.php]:
```suggestion
     * @param string|null $excludedFieldname
```

File: bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php [hunk 15]
```diff
@@ -54,6 +54,14 @@ public function __construct(DefaultMysql $model, Logger $logger)
         $this->logger = $logger;
     }
 
+    /**
+     * @param string $condition
+     * @param ?string $orderBy
+     * @param ?int $limit
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php]:
```suggestion
     * @param string|null $orderBy
     * @param int|null $limit
```

File: bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php [hunk 16]
```diff
@@ -184,6 +204,13 @@ public function loadGroupByRelationValues($fieldname, $condition, $countValues =
         }
     }
 
+    /**
+     * @param string $condition
+     * @param ?string $orderBy
+     * @param ?int $limit
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php]:
```suggestion
     * @param string|null $orderBy
     * @param int|null $limit
```

File: bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php [hunk 17]
```diff
@@ -111,6 +115,11 @@ public function getTokenList()
         return $this->getData();
     }
 
+    /**
+     * @param int $seriesId
+     * @param ?array $params
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php]:
```suggestion
     * @param array|null $params
```

File: bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php [hunk 18]
```diff
@@ -166,6 +175,11 @@ public static function getCodes($seriesId, $params)
         return $codes;
     }
 
+    /**
+     * @param int $usages
+     * @param ?int $seriesId
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php]:
```suggestion
     * @param int|null $seriesId
```

File: bundles/EcommerceFrameworkBundle/IndexService/Getter/TagsGetter.php [hunk 19]
```diff
@@ -25,6 +25,11 @@ class TagsGetter implements GetterInterface
 {
     use OptionsResolverTrait;
 
+    /**
+     * @param object $object
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Getter/TagsGetter.php]:
```suggestion
     * @param object $element
```

**Comment by fashxp** [comment]
@blankse could you please double check that PR, thx. 
Then we would merge that first to 10.6, then merge 10.6 to 11.x and then merge #13869

File: bundles/EcommerceFrameworkBundle/CartManager/AbstractCartCheckoutData.php [hunk 20]
```diff
@@ -55,6 +55,11 @@ public function getCartId()
 
     abstract public function save();
 
+    /**
+     * @param string $key
+     * @param int|string $cartId
+     * @return ?AbstractCartCheckoutData
```
Comment (mcop1) [inline bundles/EcommerceFrameworkBundle/CartManager/AbstractCartCheckoutData.php]:
```suggestion
     * @return AbstractCartCheckoutData|null
```

File: bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData.php [hunk 21]
```diff
@@ -30,6 +30,11 @@ public function save()
         $this->getDao()->save();
     }
 
+    /**
+     * @param string $key
+     * @param int|string $cartId
+     * @return ?AbstractCartCheckoutData
```
Comment (mcop1) [inline bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData.php]:
```suggestion
     * @return AbstractCartCheckoutData|null
```

File: bundles/EcommerceFrameworkBundle/CartManager/SessionCartCheckoutData.php [hunk 22]
```diff
@@ -24,6 +24,11 @@ public function save()
         throw new \Exception('Not implemented, should not be needed for this cart type.');
     }
 
+    /**
+     * @param string $key
+     * @param int|string $cartId
+     * @return ?AbstractCartCheckoutData
```
Comment (mcop1) [inline bundles/EcommerceFrameworkBundle/CartManager/SessionCartCheckoutData.php]:
```suggestion
     * @return AbstractCartCheckoutData|null
```

File: bundles/EcommerceFrameworkBundle/DependencyInjection/ServiceLocator/AssortmentTenantAwareServiceLocator.php [hunk 23]
```diff
@@ -19,6 +19,9 @@
 
 abstract class AssortmentTenantAwareServiceLocator extends TenantAwareServiceLocator
 {
+    /**
+     * @return ?string
```
Comment (mcop1) [inline bundles/EcommerceFrameworkBundle/DependencyInjection/ServiceLocator/AssortmentTenantAwareServiceLocator.php]:
```suggestion
     * @return string|null
```

File: bundles/EcommerceFrameworkBundle/DependencyInjection/ServiceLocator/CheckoutTenantAwareServiceLocator.php [hunk 24]
```diff
@@ -19,6 +19,9 @@
 
 abstract class CheckoutTenantAwareServiceLocator extends TenantAwareServiceLocator
 {
+    /**
+     * @return ?string
```
Comment (mcop1) [inline bundles/EcommerceFrameworkBundle/DependencyInjection/ServiceLocator/CheckoutTenantAwareServiceLocator.php]:
```suggestion
     * @return string|null
```

File: bundles/EcommerceFrameworkBundle/DependencyInjection/ServiceLocator/TenantAwareServiceLocator.php [hunk 25]
```diff
@@ -87,5 +90,8 @@ protected function resolveTenant(string $tenant = null)
         return $this->defaultTenant;
     }
 
+    /**
+     * @return ?string
```
Comment (mcop1) [inline bundles/EcommerceFrameworkBundle/DependencyInjection/ServiceLocator/TenantAwareServiceLocator.php]:
```suggestion
     * @return string|null
```

File: lib/Targeting/Condition/Language.php [hunk 26]
```diff
@@ -84,6 +84,9 @@ public function match(VisitorInfo $visitorInfo): bool
         return false;
     }
 
+    /**
+     * @return ?string
```
Comment (mcop1) [inline lib/Targeting/Condition/Language.php]:
```suggestion
     * @return string|null
```

**Review by mcop1 — COMMENTED** [review]
Apparently I need to add a comment, so we should use `|null` in php doc annotations

File: bundles/EcommerceFrameworkBundle/IndexService/Config/Definition/Attribute.php [hunk 27]
```diff
@@ -154,6 +154,11 @@ public function getOptions(): array
         return $this->options;
     }
 
+    /**
+     * @param string $name
+     * @param mixed $defaultValue
+     * @return mixed|null
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/Config/Definition/Attribute.php]:
```suggestion
     * @return mixed
```

File: bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php [hunk 28]
```diff
@@ -54,6 +54,14 @@ public function __construct(DefaultMysql $model, Logger $logger)
...
 
+    /**
+     * @param string $condition
+     * @param string|null $orderBy
+     * @param int|null $limit
+     * @param int $offset
+     * @return array
+     * @throws \Doctrine\DBAL\Exception
+     */
     public function load($condition, $orderBy = null, $limit = null, $offset = null)
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php]:
```suggestion
    public function load($condition, $orderBy = null, $limit = null, $offset = 0)
```

File: bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php [hunk 29]
```diff
@@ -184,6 +204,13 @@ public function loadGroupByRelationValues($fieldname, $condition, $countValues =
...
     }
 
+    /**
+     * @param string $condition
+     * @param string|null $orderBy
+     * @param int|null $limit
+     * @param int $offset
+     * @return int
+     */
     public function getCount($condition, $orderBy = null, $limit = null, $offset = null)
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/IndexService/ProductList/DefaultMysql/Dao.php]:
```suggestion
    public function getCount($condition, $orderBy = null, $limit = null, $offset = 0)
```

File: bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php [hunk 30]
```diff
@@ -184,6 +198,10 @@ public static function getCountByUsages($usages = 1, $seriesId = null)
         }
     }
 
+    /**
+     * @param int $seriesId
+     * @return bool|int
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php]:
```suggestion
     * @return int|false
```

File: bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php [hunk 31]
```diff
@@ -198,6 +216,10 @@ public static function getCountBySeriesId($seriesId)
...
```
Comment (blankse) [inline bundles/EcommerceFrameworkBundle/VoucherService/Token/Listing.php]:
```suggestion
     * @return int|false
```

File: lib/Image/Adapter/Imagick.php [hunk 32]
```diff
@@ -922,7 +922,11 @@ public function sepia()
     }
 
     /**
-     * {@inheritdoc}
+     * @param float $radius
+     * @param float $sigma
+     * @param float $amount
+     * @param float $threshold
+     * @return static
```
Comment (blankse) [inline lib/Image/Adapter/Imagick.php]:
```suggestion
     * @return $this
```

File: models/DataObject/ClassDefinition/Data/BooleanSelect.php [hunk 33]
```diff
@@ -374,6 +379,10 @@ public function getNoLabel()
         return $this->noLabel;
     }
 
+    /**
+     * @param string $noLabel
+     * @return static
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/BooleanSelect.php]:
```suggestion
     * @return $this
```

File: models/DataObject/ClassDefinition/Data/BooleanSelect.php [hunk 34]
```diff
@@ -390,6 +399,10 @@ public function getEmptyLabel()
         return $this->emptyLabel;
     }
 
+    /**
+     * @param string $emptyLabel
+     * @return static
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/BooleanSelect.php]:
```suggestion
     * @return $this
```

File: models/Document/Hardlink/Wrapper.php [hunk 35]
```diff
@@ -216,8 +216,10 @@ public function getSourceDocument(): ?Document
     /**
      * @param Document $sourceDocument
```
Comment (blankse) [inline models/Document/Hardlink/Wrapper.php:217]:
```suggestion
     * @param Document $sourceDocument
     * @return $this
```

File: models/Document/Hardlink/Wrapper/WrapperInterface.php [hunk 36]
```diff
@@ -22,6 +22,7 @@ interface WrapperInterface extends ElementInterface
 {
     /**
      * @param Document $hardLinkSource
+     * @return static
```
Comment (blankse) [inline models/Document/Hardlink/Wrapper/WrapperInterface.php]:
```suggestion
     * @return $this
```

File: models/Document/Hardlink/Wrapper/WrapperInterface.php [hunk 37]
```diff
@@ -32,6 +33,7 @@ public function getHardLinkSource();
 
     /**
      * @param Document $sourceDocument
+     * @return static
```
Comment (blankse) [inline models/Document/Hardlink/Wrapper/WrapperInterface.php]:
```suggestion
     * @return $this
```

**Comment by blankse** [comment]
This setter has void return type and all other setters in this class has return $this. Should we also unify it?

https://github.com/pimcore/pimcore/blob/67654a9eec53ee597573fd1544f6fffbc6160bf5/models/Document/DocType.php#L259

**Comment by blankse** [comment]
I fixed also types in Pimcore\Tool\Console https://github.com/pimcore/pimcore/pull/13869/commits/9962ea4e94262a8da92aaa1086b46ca377824bbc
Should we adopt the change here as well?

**Comment by fashxp** [comment]
> This setter has void return type and all other setters in this class has return $this. Should we also unify it?
> 
> https://github.com/pimcore/pimcore/blob/67654a9eec53ee597573fd1544f6fffbc6160bf5/models/Document/DocType.php#L259

lets do that in a separate PR in 11.x

**Comment by fashxp** [comment]
> I fixed also types in Pimcore\Tool\Console [9962ea4](https://github.com/pimcore/pimcore/commit/9962ea4e94262a8da92aaa1086b46ca377824bbc) Should we adopt the change here as well?

`Pimcore/Tool/Console` is a final class, so not really necessary there. Reason for updating php-doc in 10.6 is to allow developers adapting their classes extending Pimcore classes accordingly before migrating to Pimcore 11.

**Comment by blankse** [comment]
@fashxp Okay, than this LGTM :)

**Pull request merged**